### DISCUSSION TITLE
link issue to a specific file in repo

### DISCUSSION THREAD

**Discussion by mark-corsillo** [community/community#38182] [Projects and Issues]
It would be great if there was an option to open an issue with a specific file, so that when opening a file in GitHub you could see all issues associated with it, or see a list of issues associated with any files changed in a PR.

I'd like to be able to use issues to note nice-to-have changes to various scripts in the repo, things that don't necessarily warrant a dedicated PR but should be done next time someone is working on that file for something else.

**Comment by david-engelmann** [comment]
Typically, I use comments for that. However, there isn't anything stopping you from linking files in Issues.

[Here is an example with one of my personal projects](https://github.com/david-engelmann/DoccanoClient.jl/issues/11)

**Reply by mark-corsillo** [reply]
comments don't work well for this in protected repos, though. in a protected repo, i'd have to go into the file to add a comment, then create a branch, commit the changed file, and create a PR just to add a todo comment. this defeats the purpose, because all i'm trying to do is leave a note for the next person who's working on the file to make a minor edit. if i already have to go through the whole rigamarole of commits/PRs, i might as well just make the minor edit in the file myself rather than adding the comment.

**Reply by mark-corsillo** [reply]
also, putting links to the file in the issue is helpful but it still requires you to pull up the issues. what i'm suggesting would let you see linked issues from the file itself, or from PRs that include changes to the linked files.

**Comment by Donnerstagnacht** [comment]
I think with the raising use of ai tools, this would be a super cool feature.

Ideally, one could reference a file by path from the root folder.

Then one could pull the issue with an mcp to the agent which can transform the link to a local path and would cut down agentic search costs.

I think github should be able to render that path as link in the web gui and as path in a local view, or the agent could transfer the link to a path.

Bonus would be many features like showng all issues cinnected to a file as mentioned.

 A larger issue is that the link/path will breaj once files are renamed or moved, but I think that woupd be an acceptable trade off.

UX could be similiar like in vs code copilot chat, somethink like `#path` or another shortcut since that hashtag is already used.

Since copilot indexes repositories remotly, github should be able to open that api/knowledge to the create issue process.
